zeebe/gateway/src/main/java/io/camunda/zeebe/gateway/impl/broker/request/role/BrokerRoleDeleteRequest.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.gateway.impl.broker.request.role;
+
+import io.camunda.zeebe.broker.client.api.dto.BrokerExecuteCommand;
+import io.camunda.zeebe.protocol.Protocol;
+import io.camunda.zeebe.protocol.impl.record.value.authorization.RoleRecord;
+import io.camunda.zeebe.protocol.record.ValueType;
+import io.camunda.zeebe.protocol.record.intent.RoleIntent;
+import org.agrona.DirectBuffer;
+
+public class BrokerRoleDeleteRequest extends BrokerExecuteCommand<RoleRecord> {
+  private final RoleRecord requestDto = new RoleRecord();
+
+  public BrokerRoleDeleteRequest() {
+    super(ValueType.ROLE, RoleIntent.DELETE);
+    setPartitionId(Protocol.DEPLOYMENT_PARTITION);
+  }
```

Comment:
❓ In other instances where we work with existing entities we provide the key on the request [REF](https://github.com/camunda/camunda/blob/main/zeebe/gateway/src/main/java/io/camunda/zeebe/gateway/impl/broker/request/BrokerRoleUpdateRequest.java#L23C2-L23C25) could you check if we also should be setting that here? :)

Reply:
I believe we don't have to (for example `BrokerUserDeleteRequest` doesn't set the key in the constructor). I think it's a better pattern to provide the required key there though, so I changed it anyway.